### DISCUSSION TITLE
Display player dynamically

### DISCUSSION THREAD

**Discussion by m90att** [punxaphil/custom-sonos-card#519] [Q&A]
Hi, I've searched but can't find the answer to this question.  I know it is possible to set a player so it is the one that is displayed by default.  However, if that speaker isn't playing and another one is, I'd like the playing one to display by default.  An example use case is that when I'm at work in the study, the study player should be default.  I often have to play and pause it due to meetings.  In the evening when I'm not working, I'd like whichever of the lounge, kitchen, bathroom player to be default if it's playing.  I know there may have to be something that will prioritise one player over another if more than one are playing. 

Is this possible?

**Comment by punxaphil** [comment]
Not supported today, but I could add it. Just want to make sure I understand the logic you are after:
Something like this?
```typescript
if (selectedPlayer) {
      if (selectedPlayer.isNotPlaying() && config.overrideSelectedEntityIfOthersArePlaying && playingPlayer) {
        return playingPlayer;
      } else {
        return selectedPlayer;
      }
}
if (playingPlayer) {
  return playingPlayer;
}
return firstPlayerInList;
```

**Reply by m90att** [reply]
Hi, thanks for the swift response.  Yes, that looks good to me. Is there anywhere to configure the list of players priority, which I'm assuming the firstPlayerInList refers to?

**Reply by punxaphil** [reply]
Good point. I updated the card to respect the order of the entities in config.
https://github.com/punxaphil/custom-sonos-card/releases/tag/v7.21.0


---
PunxaPhil
[![image](https://github.com/user-attachments/assets/cb2e240e-f9ca-4900-a425-b8089216c1ee)](https://patreon.com/PunxaPhil) [![image](https://github.com/johanfrick/custom-sonos-card/assets/26582898/9bebc1f6-6bda-4cbd-b596-50d3d2f712de)](https://www.buymeacoffee.com/punxaphil)

**Comment by m90att** [comment]
Hi, this is fantastic. It seems to be working well but as I'm away for a
couple of days I'll test it properly when i get home. Thank you for making
the change.
